crates/re_viewer_context/src/space_view/auto_spawn_heuristic.rs

```diff
@@ -0,0 +1,16 @@
+/// Heuristic result used to determine if a Space View with a given class should be automatically spawned.
+///
+/// The details of how this is interpreted are up to the code determining candidates and performing
+/// Space View spawning.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AutoSpawnHeuristic {
+    /// Always spawn a Space View with this class if it is a candidate.
+    AlwaysSpawn,
+
+    /// Never spawn a Space View with this class if it is a candidate.
+    /// This means that the candidate can only be created manually.
+    NeverSpawn,
+
+    /// If there's several candidates for the same root, spawn only one with the highest score.
+    SpawnClassWithHighestScoreForRoot(f32),
```

Comment:
Right now there is an implicit assumption that all SpaceViews which return `SpawnClassWithHighestScoreForRoot` are in some sense equivalent.  This might be true today but ends up being a bit limiting.

Adding something like a `equivalency_id` along with the score would let us be a bit more explicit. SpaceView2D and SpaceView3D could both then use the "spatial" equivalency_id, but a different view could still use SpawnClassWithHighestScoreForRoot if there was an entirely different view that should *also* be included due to a non-spatial type being present in the space.

Reply:
left a comment for later to do that